types/bluebird/bluebird-tests.ts

```diff
@@ -373,6 +373,30 @@ fooProm = fooProm.catch(CustomError, reason => {
 	fooProm = fooProm.catch(CustomError1, CustomError2, CustomError3, error => {});
 	fooProm = fooProm.catch(CustomError1, CustomError2, CustomError3, CustomError4, error => {});
 	fooProm = fooProm.catch(CustomError1, CustomError2, CustomError3, CustomError4, CustomError5, error => {});
+
+	const booPredicate1 = (error: CustomError1) => true;
+	const booPredicate2 = (error: [number]) => true;
+	const booPredicate3 = (error: string) => true;
+	const booPredicate4 = (error: Object) => true;
+	const booPredicate5 = (error: any) => true;
+
+	fooProm = fooProm.catch(booPredicate1, error => {});
+	fooProm = fooProm.catch(booPredicate1, booPredicate2, error => {});
+	fooProm = fooProm.catch(booPredicate1, booPredicate2, booPredicate3, error => {});
+	fooProm = fooProm.catch(booPredicate1, booPredicate2, booPredicate3, booPredicate4, error => {});
+	fooProm = fooProm.catch(booPredicate1, booPredicate2, booPredicate3, booPredicate4, booPredicate5, error => {});
+
+	const booObject1 = new CustomError1();
+	const booObject2 = [400, 500];
+	const booObject3 = ["Error1", "Error2"];
+	const booObject4 = {code: 400};
+	const booObject5: any = null;
+
+	fooProm = fooProm.catch(booObject1, error => {});
+	fooProm = fooProm.catch(booObject1, booObject2, error => {});
+	fooProm = fooProm.catch(booObject1, booObject2, booObject3, error => {});
+	fooProm = fooProm.catch(booObject1, booObject2, booObject3, booObject4, error => {});
+	fooProm = fooProm.catch(booObject1, booObject2, booObject3, booObject4, booObject5, error => {});
```

Comment:
Ah, okay. I guess the only downside in this case is that you might get auto-complete options for the static keys of `CustomError1` instead of the keys of an instance of `CustomError1`. Is that correct?